public/scripts/project/project.js

```diff
@@ -103,6 +104,18 @@ define(function(require) {
 
     var metadataLocation = _user && _anonymousId ? Path.join(Constants.ANONYMOUS_USER_FOLDER, _anonymousId.toString()) : getRoot();
 
+    // First, lock the project or bail
+    var locked = Lock.requestLock(_id);
+    if (!locked) {
+      // Abort!
+      window.location = "/projects";
+      return;
+    }
+
+    // Remove the lock when the user navigates away from the page, or closes
+    // the tab.
+    window.addEventListener("beforeunload", Lock.removeLock.bind(Lock, _id));
```

Comment:
Why do you bind here? `Lock` isn't an instance of a constructor and you don't use `this inside `removeLock`